src/rulesets/rest/2022-05-25/specification-rules.ts

```diff
@@ -1,8 +1,45 @@
 import { RuleError, Ruleset, SpecificationRule } from "@useoptic/rulesets-base";
-import { pascalCase } from "change-case";
+import { dotCase, pascalCase, snakeCase } from "change-case";
 import { links } from "../../../docs";
 import { stabilityKey } from "./constants";
 
+interface componentName {
+  localName: string;
+  localProp?: string;
+  ns?: string;
+}
+
+/**
+ * Decode the component name from a possibly namespace and property-qualified OpenAPI component name.
+ *
+ * @param name An OpenAPI component name
+ * @returns Tuple containing the local name (without namespace) and its namespace prefix (undefined if not namespaced).
+ */
+const decodeComponentName = (name: string): componentName => {
+  const parts = name.split(".");
+  if (parts.length > 1) {
+    const lastPart = parts[parts.length - 1];
+    if (lastPart === snakeCase(lastPart)) {
+      // Component is derived from a Typespec model property, of the form namespace.ModelName.property_name.
+      // This is the case for shared parameter or header components, for example.
+      return {
+        localName: parts[parts.length - 2],
+        localProp: lastPart,
+        ns: parts.slice(0, parts.length - 2).join("."),
+      };
+    } else {
+      // Component is derived from a Typespec model, of the form namespace.ModelName.property_name.
```

Comment:
Not sure this is accurate given the conditions needed to reach this point: length > 1 && lastPart != snakeCase

Reply:
Oh, it's a comment typo. 🤦 Got it!